src/main/java/io/tus/java/client/TusClient.java

```diff
@@ -19,6 +20,7 @@ public class TusClient {
     public static final String TUS_VERSION = "1.0.0";
 
     private URL uploadCreationURL;
+    private Proxy uploadProxy = Proxy.NO_PROXY;
```

Comment:
I am concerned about the default value of `NO_PROXY`. If I read https://docs.oracle.com/javase/7/docs/api/java/net/URL.html#openConnection%28java.net.Proxy%29 correctly, it will also ignore system-wide proxy settings. This would be a breaking change because tus-java-client respects these settings by default right now.

Can you comment if I am wrong here or modify your PR so that this behavior of tus-java-client is not changed?

Reply:
Yeah it seems your right, I looked into the implementation and the 'magic' only happens when proxy == null. Gonna change that so it's optional.